### DISCUSSION TITLE
Integrating SmartDry into Theengs

### DISCUSSION THREAD

**Discussion by titelies** [theengs/app#47] [Q&A]
I was in fact near the sensor for about 10 minutes. I set the dryer for 10 mins and I watched as the Model # of my SmartDry appeared in the Theengs device browser. I then switched to sensors and saw it appear but it originally showed Unsynced I believe. After 2 mins had passed on the dryer, it showed up as synced but none of the Humidity, Shake, Temp, Volt, and Wake values ever changed. My phone was sitting above the dryer for the remainder of the 8 mins.

**Comment by 1technophile** [comment]
Could share please a screenshot of what you see for the sensor, main page and sensor page?

**Reply by titelies** [reply]
Sure... ![Screenshot_20221112-170947.png](https://user-images.githubusercontent.com/16294530/201499545-7874af30-51a4-4777-8d2e-9c94cbfaabf3.png)

**Reply by titelies** [reply]
![Screenshot_20221112-170958.png](https://user-images.githubusercontent.com/16294530/201499549-2229c66f-799b-4803-b64e-3730696c1438.png)

**Reply by titelies** [reply]
![Screenshot_20221112-171408.png](https://user-images.githubusercontent.com/16294530/201499593-3a67f7bf-8e39-4dee-a401-aea22ffae475.png)

**Reply by 1technophile** [reply]
Regarding the app decoding so as to see what's going on, would you mind installing the tool below - toolBLEx:
https://github.com/emericg/toolBLEx/actions/runs/3298215330

And share with us a screenshot with the SmartDry and advertisement tab selected:
![image](https://user-images.githubusercontent.com/12672732/201537358-faec2662-ff90-4db9-abab-a43886b5962b.png)

You can hide the other sensors

**Reply by titelies** [reply]
Can I install this on my laptop running MX Linux or does it need to be installed on the Pi for Home Assistant.?

**Reply by 1technophile** [reply]
You can install it on your laptop or desktop also.

**Reply by titelies** [reply]
I installed it on my laptop and it appears along with the other Bluetooth devices but it's grayed out so I can't click the Advertisement tab to report any results.

**Reply by titelies** [reply]
I did just launch the Theengs app and it's reporting details today... ![Screenshot_20221113-133924.png](https://user-images.githubusercontent.com/16294530/201538669-a533ca60-cf91-42ec-9fb3-8608c20a57d7.png)

**Reply by 1technophile** [reply]
Cool, are you using Android or iOS ?

**Reply by titelies** [reply]
Pixel 7 Pro running Android 13

**Reply by 1technophile** [reply]
To maximize the scan frequency you can activate the background mode in settings every 5 min and use these tips:
https://app.theengs.io/use/use.html#android-10

**Reply by titelies** [reply]
Is it normal for VPN on my mobile phone to effect the results in Theengs even when I have Allow local networks to be routed outside the VPN and I also enabled Split Tunneling for the Theengs app specifically?

**Reply by titelies** [reply]
> To maximize the scan frequency you can activate the background mode in settings every 5 min and use these tips:
> https://app.theengs.io/use/use.html#android-10

I had followed those steps already when I first installed the Theengs app.

**Reply by 1technophile** [reply]
Do you mean that when the VPN is active, Theengs app is not pushing data to the MQTT broker  ?

**Reply by titelies** [reply]
With VPN on... ![Screenshot_20221113-140458.png](https://user-images.githubusercontent.com/16294530/201539656-6e6945e8-c484-4cfa-9f46-0b962e771ac0.png)

**Reply by titelies** [reply]
Wondering if that's why I wasn't getting any results yesterday... Don't remember if I had it or not when testing.

**Reply by titelies** [reply]
Sorry for the confusion... Maybe it's distance? Does the phone and Theengs app need to be near the dryer to Sync and return results?

**Reply by 1technophile** [reply]
You should be able to get results 5m or 15feet away. More with line of sights.

**Reply by titelies** [reply]
Ok that seems accurate even with the VPN enabled.

**Reply by 1technophile** [reply]
Is all working fine now @titelies?

**Reply by titelies** [reply]
> Is all working fine now @titelies?

I'm happy 😊! Thanks for ur support.

**Reply by 1technophile** [reply]
Do you mind if I use your screenshots of the SmartDry in the app ( I will remove any id from it)  ?

**Reply by titelies** [reply]
> Do you mind if I use your screenshots of the SmartDry in the app ( I will remove any id from it)  ?

No problem.

**Comment by 1technophile** [comment]
Thanks, are you familiar with MQTT?

**Reply by titelies** [reply]
I am not. This would be my first setup for use within MQTT with Home Assistant

**Comment by 1technophile** [comment]
On which kind of server is HA installed?

**Reply by titelies** [reply]
HA is installed on a Pi4-64

**Reply by 1technophile** [reply]
How far is it from the SmartDry?

**Reply by titelies** [reply]
Approximately 30 feet away

**Reply by 1technophile** [reply]
Ok let's try to get it from the pi.

1) Install MQTT integration
https://www.home-assistant.io/integrations/mqtt/

2) Create a new user in HA like `mqttuser`

3) Install Theengs gateway addon and add the user/password from step 2 into the configuration tab
https://gateway.theengs.io/install/install.html#install-theengs-gateway-as-an-add-on-in-home-assistant

4) download mqtt explorer, connect it to your server by entering, ip of the sever, username and password from step 2
http://mqtt-explorer.com/

Check if you see messages coming from the SmartDry into mqtt explorer

**Reply by titelies** [reply]
Showing up in MQTT Explorer Add-on ![PXL_20221113_013122573.jpg](https://user-images.githubusercontent.com/16294530/201501265-06b4dc33-7c53-4bf9-bcff-8aad8d9d9861.jpg)

**Reply by 1technophile** [reply]
Cool, so now if you go into HA, Settings-> Device and Services -> Devices you can check if the SmartDry appears and add it to a dashboard.

**Reply by titelies** [reply]
It does.... Shows Humidity and Temperature sensors. Thank you for the assistance!

**Reply by 1technophile** [reply]
You may want to reduce `TIME_BETWEEN` to 1s to catch all the data sent by the sensor

**Reply by titelies** [reply]
After a restart, more sensors appeared... ![PXL_20221113_015730853.jpg](https://user-images.githubusercontent.com/16294530/201501859-90655f0e-dcd4-4655-b809-6c9c0527aca9.jpg)
